akka-cluster-sharding/src/test/scala/akka/cluster/sharding/LeastShardAllocationStrategySpec.scala

```diff
@@ -129,6 +129,119 @@ class LeastShardAllocationStrategySpec extends AkkaSpec {
+    }
+
+    "rebalance from region with most number of shards [5, 0, 0], rebalanceThreshold=1, rebalanceFactor=0.5" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 0,
+        rebalanceFactor = 0.5)
+      val allocations = createAllocations(aCount = 5)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001", "002"))
+    }
+
+    "rebalance from region with most number of shards [6, 0, 0], rebalanceThreshold=1, rebalanceFactor=0.5" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 0,
+        rebalanceFactor = 0.5)
+      val allocations = createAllocations(aCount = 6)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001", "002", "003"))
+    }
+
+    "rebalance from region with most number of shards [6, 0, 0], rebalanceThreshold=1, rebalanceFactor=0.2" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 0,
+        rebalanceFactor = 0.2)
+      val allocations = createAllocations(aCount = 6)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001"))
+    }
+
+    "rebalance from region with most number of shards [6, 0, 0], rebalanceThreshold=1, rebalanceFactor=0.4" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 0,
+        rebalanceFactor = 0.4)
+      val allocations = createAllocations(aCount = 6)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001", "002"))
+    }
+
+    "rebalance from region with most number of shards [6, 0, 0], rebalanceThreshold=1, rebalanceNumber=2, rebalanceFactor=0.5" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 2,
+        rebalanceFactor = 0.5)
+      val allocations = createAllocations(aCount = 6)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001", "002"))
+    }
+
+    "rebalance from region with most number of shards [6, 0, 0], rebalanceThreshold=1, rebalanceNumber=2, rebalanceFactor=0.2" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 2,
+        rebalanceFactor = 0.2)
+      val allocations = createAllocations(aCount = 6)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001"))
+    }
+
```

Comment:
More likely scenarios are [N, N, 0] when a new node is added.

Reply:
True, that said, because the allocation strategy only considers the region with the most number of shards, what the other region contains is irrelevant.